include/swift/AST/DiagnosticsCommon.def

```diff
@@ -95,6 +95,9 @@ NOTE(profile_read_error,none,
 ERROR(generic_signature_not_minimal,none,
       "generic requirement '%0' is redundant in %1", (StringRef, StringRef))
 
+WARNING(extension_redundant_requirement,none,
+"redundant requirement in extension", ())
```

Comment:
The diagnostic would be still more useful, I think, if it specified which requirement is redundant—see the error above it for an example of the wording.

It’d also be nice to have a very succinct explanation as to why it’s redundant (is it a subclass? is it a less refined protocol?).

Also, nit: see the rest of the file for indentation style.

Reply:
Thanks @xwu, I have updated the diagnostic message and fixed the indention.